Update runtime-spec to v1.2.1

kiashok opened the pull request:
Update runtime-spec to v1.2.1

Full changelist can be found here: https://github.com/opencontainers/runtime-spec/compare/v1.2.0...v1.2.1

kiashok commented:
cc @jsturtevant

mikebrow reviewed (approved):
LGTM
noting we are not amending our contrib/seccomp profile code for the arch changes; or making changes for the cpu affinity additions  

Maybe open up TODO issues to investigate/update for these changes?

kiashok commented:
> LGTM noting we are not amending our contrib/seccomp profile code for the arch changes; or making changes for the cpu affinity additions
> 
> Maybe open up TODO issues to investigate/update for these changes?

what seccomp profile changes are you referring to? This runtime-spec version will help us implement cpu affinity changes for windows on hcsshim side. containerd will only be routing the values through. Am I missing some other change you are referring to?

estesp reviewed (approved):
LGTM, but needs a rebase to merge

mikebrow commented:
> > LGTM noting we are not amending our contrib/seccomp profile code for the arch changes; or making changes for the cpu affinity additions
> > Maybe open up TODO issues to investigate/update for these changes?
> 
> what seccomp profile changes are you referring to? This runtime-spec version will help us implement cpu affinity changes for windows on hcsshim side. containerd will only be routing the values through. Am I missing some other change you are referring to?

noting the new arch entries .. it reminded me that each arch entry needs a mapping to it's seccomp defaults.
see https://github.com/containerd/containerd/blob/main/contrib/seccomp/seccomp_default.go I had to do a lot of fishing around to find everyone's best default set.

kiashok commented:
> Needs to resolve the conflict in go.mod cc: @kiashok

updated!

kiashok commented:
> LGTM, but needs a rebase to merge

updated the PR

jsturtevant commented on containerd.exe:
Should this be here?

jsturtevant replied:
@estesp

kiashok replied:
removed it

estesp replied:
Thanks! 😅  Moving too quickly and didn't see the new file. Thanks for fixing @kiashok

djdongjin commented:
@estesp I removed the PR from merge queue (🤔surprised reviewer can remove given cannot add).

@kiashok I think this need to be removed https://github.com/containerd/containerd/pull/11460/files#diff-202f0f86324ce87e6d8b36a1f8e786e6d963a717090864ce7bf985c35f920c66

kiashok commented:
> @estesp I removed the PR from merge queue (🤔surprised reviewer can remove given cannot add).
> 
> @kiashok I think this need to be removed https://github.com/containerd/containerd/pull/11460/files#diff-202f0f86324ce87e6d8b36a1f8e786e6d963a717090864ce7bf985c35f920c66

updated it. Could you please add to the merge queue once the CIs are green? Thanks!

samuelkarp commented:
> noting the new arch entries .. it reminded me that each arch entry needs a mapping to it's seccomp defaults.
see [main/contrib/seccomp/seccomp_default.go](https://github.com/containerd/containerd/blob/main/contrib/seccomp/seccomp_default.go) I had to do a lot of fishing around to find everyone's best default set.

Sounds like we need a follow-up PR to handle SCMP_ARCH_LOONGARCH64, SCMP_ARCH_M68K, SCMP_ARCH_SH, and SCMP_ARCH_SHEB if we want to support those architectures.  cc @zhaixiaojuan to update https://github.com/containerd/containerd/pull/11215 once this is merged.

kiashok commented:
> > noting the new arch entries .. it reminded me that each arch entry needs a mapping to it's seccomp defaults.
> > see [main/contrib/seccomp/seccomp_default.go](https://github.com/containerd/containerd/blob/main/contrib/seccomp/seccomp_default.go) I had to do a lot of fishing around to find everyone's best default set.
> 
> Sounds like we need a follow-up PR to handle SCMP_ARCH_LOONGARCH64, SCMP_ARCH_M68K, SCMP_ARCH_SH, and SCMP_ARCH_SHEB if we want to support those architectures. cc @zhaixiaojuan to update #11215 once this is merged.

Thanks for pointing out this PR. I was just about to open up an issue to track @mikebrow 's suggestion.
